### ISSUE TITLE
Phone freezes when tapping on Signal notification with image

### ISSUE THREAD

**Issue by mlsxlist** [signalapp/Signal-Android#11828]
- [X] I have searched open and closed issues for duplicates
- [X] I am submitting a bug report for existing functionality that does not work as intended
- [X] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [X] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When a new Signal message that contains media like an image arrives a notification is shown. Tapping on the notification makes the complete phone to freeze. Then an Android system dialog is shown that asks it I want to wait or exit.

### Steps to reproduce
- let someone send a signal message that contains an image to you
- once the notification is shown tap on it

**Actual result:** The complete Android phone freezes and a Android system UI dialog asks if I want to wait or exit.If I wait for my phone Android UI starts after a long time. If I exit Android UI is restarted
**Expected result:** Signal should display the message

### Device info
**Device:** HUAWEIVOG-L29
**Android version:** 10.0
**Signal version:** 5.27.13

**Comment by mlsxlist** [comment]
Here another debug log from a colleague:

https://debuglogs.org/1321885a5aeed86332e019f4bba6eaf1e3a75ce8af26c4e9a7447d750c94d153

**Comment by Simsala91** [comment]
Having the exact same issue.

Also on HUAWEI VOG-L29, Android 10, Signal 5.27.13, so this might be connected with how EMUI handles notifications. 

I've been having this issue for a while now, at least for the last 3 months.

As a hotfix I changed signal settings so notifications only show the name of the contact.

**Comment by mlsxlist** [comment]
My hotfix is not to immediately tap on the notification when it comes up.

**Comment by wertarbyte** [comment]
I'm seeing the same issue on a Motorola One, Android 10 as well.

**Comment by mlsxlist** [comment]
> I'm seeing the same issue on a Motorola One, Android 10 as well.

That is interesting. So far I only knew people with Huawei P30 Pro who experienced this issue. Can you please post a debug log.

**Comment by mwaterbu** [comment]
Also having this issue with a Motorola Moto X4 (Android 9).
I do not need to immediately tap the notification for it to freeze, or even see the notification at all. If my phone is locked, and I receive this type of message, the System UI locks up, making it nearly impossible to even unlock my phone (~20% chance to be able to unlock after 5+ minutes waiting, otherwise, have to hard reboot).

**Comment by mlsxlist** [comment]
@mwaterbu can you please add a debug log. Thanks.

**Comment by mlsxlist** [comment]
Yes, it still happens and is very annoying, as the complete phone often has to be restarted. Would appreciate it a Signal dev could recreate this with a Huawei P30 Pro.

**Comment by FakeFurball** [comment]
Running into the same issue on a pixel 2xl, and it's getting worse with time.  It used to be if I tapped the notification, now it's if the screen is on and I get the notification, the system ui completely freezes and often requires a phone restart to become usable again.

**Comment by mlsxlist** [comment]
Please add a debuglog.

**Comment by kaiduerkop** [comment]
Same issue here.
On every Message with an attached Image the Phone is going crazy.

Huawei P30 Pro

**Comment by clauz9** [comment]
fixed with: https://github.com/signalapp/Signal-Android/commit/8f2c5d43dfa61c59dd377c2c3e6f20b00e82ffc7

**Comment by alex-signal** [comment]
This should be resolved in the latest beta.

**Comment by mlsxlist** [comment]
Thanks. Will test once 5.38.1 ist available in Playstore.

**Comment by mlsxlist** [comment]
https://github.com/signalapp/Signal-Android/commit/8f2c5d43dfa61c59dd377c2c3e6f20b00e82ffc7

**Comment by 1Yanik3** [comment]
using version 5.38.3, have the same issue (System UI Freezes when notification is received)

**Comment by cody-signal** [comment]
@1Yanik3 can you provide a debuglog after encountering the issue?

**Comment by acx0** [comment]
@cody-signal just ran into this bug today, although in my case my phone will freeze without any interaction upon an image being received (i.e. phone won't even respond when trying to unlock it / takes up to 10 seconds for screen to unlock)

some other notes:
- I generally have signal desktop open as well so once I view the message there sometimes my phone will eventually begin responding again without me having to do a hard reboot, as was the case today

debug log: https://debuglogs.org/android/5.39.3/31e44583f0983e171e08d3e6c8a9fd2cbb359612757f1c709b7fa4a4bda8370f
- the offending message in this case has the following details in the app: (times are in EDT)
```
Sent | Today 1:06 PM (1654189605210)
-- | --
Received | Today 1:08 PM (1654189699472)
```

**Comment by mlsxlist** [comment]
Still happens with 5.39.3

https://debuglogs.org/android/5.39.3/bc211fe60e0dead64810ef9911bcfd0d9254707b542be2632cb5e5ba66776555

**Comment by Chris78** [comment]
Same here. Glad it's not only me to experience this, since I'm on LineageOS 18.1-20220512-NIGHTLY-bacon (Android 11) on an older OnePlus One.

For now, I have changed the notification settings to only show the name of the sender, which hopefully helps temporarily.

**Comment by mlsxlist** [comment]
@Chris78 please add debug logs.

**Comment by Chris78** [comment]
https://debuglogs.org/android/5.40.4/d7fc02732c4bb983f050dbe502a0879c39c8cdefb0d8ab29f9d736ef7ee69ca9

**Comment by vanMacG** [comment]
I have the same problem since quite some time. Occurs on two devices (wife & mine). Problem is independant of sender. Not easy to reproduce because it occurs only sometimes...
Despite of the topic desciption "when tapping", the problem occurs also when the phone is not touched or unattended. (https://github.com/signalapp/Signal-Android/issues/11685 was closed as duplicate of this issue.)

https://debuglogs.org/android/5.44.4/2e89c40c99d67b418b00499969bc133be7e186743e7856bf478fadcc8bbbd591

Problematic message with image received at `2022-08-10 11:11:11.282`

**Comment by Chris78** [comment]
btw. I have had no more freezes, since I set the notification settings to only show the sender.
However it's not very satisfying if you cannot read a message within the notification...

A notification option to show only pure text messages in the notification and only the sender for media messages (sent images, videos) might be a good compromise until this is fixed.

**Comment by mlsxlist** [comment]
I still have hope that this will be fixed at some point in time. The annoying crashes still happen.

**Comment by cody-signal** [comment]
Hi folks, can you tell me the rough size of the images that are causing the problem or does it happen with every image? If possible sending a range of image sizes or types and letting me know if anything is reproducible about the data being used would be super helpful.

**Comment by mlsxlist** [comment]
The issued just happened for me. The image has a resolution of 3648x2736 pixel. Signal ist configured to send media with high qualitiy.

https://debuglogs.org/android/5.44.5/627598b7f77c91a42713b16995acc8a44f7cb3400db5ce7dcce548f66fed9719

The message that caused the issue was send to a group.

**Comment by vanMacG** [comment]
> Hi folks, can you tell me the rough size of the images that are causing the problem or does it happen with every image?

Imho it can happen with every image. The last two problematic images I remember have 150KB and 750KB. But it is not reproduciable... After rebooting the phone and being sent the same image again it was no problem anymore.

I'm sorry, but this is so random...

**Comment by mlsxlist** [comment]
Got a new crash.

https://debuglogs.org/android/5.45.6/c1459147a236d2bf8afbd08103f25903e4ebd994a62d10e24a4402bb2355bfca

**Comment by daryluren** [comment]
Same problem. Also a OnePlus 5T, Oxygen 10.0.1.

I've experienced the following which may be clues or workarounds, or possibly just coincidences.

1. When the phone freezes I can go read the offending message on my desktop. Then the phone will unfreeze.
2. On some occasions plugging in the phone to charge will unfreeze it.

**Comment by mwaterbu** [comment]
I no longer have this issue after switching to LineageOS (upgrading from Android 9 to 12).
However, it is also possible that I just haven't run into it yet. I use Signal much less frequently now, because so many of my contacts left the platform after the SMS removal.

**Comment by cody-signal** [comment]
@mrphil2105 can you post an updated debuglog from Settings->Help->Debuglog?
